### PR TITLE
WIP: {tools,vis}[foss/2019a] Sphinx v1.8.3, Spyder v4.1.5 w/ Python 3.7.2

### PR DESCRIPTION
**Pull request by zarybnicky**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/s/Sphinx/Sphinx-1.8.3-foss-2019a-Python-3.7.2.eb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }),
+    ('sphinx-bootstrap-theme', '0.6.5', {
+        'checksums': ['82936109a05b84029052e7efb1756161c3b4c27ae18512a90234d50fe608cba7'],
+    }),
+    ('sphinx-gallery', '0.2.0', {
+        'checksums': ['b49356b5516cc7dab67b1b378f6bf8146fe2372ee73d5e1ea2c483a2e3f4f182'],
+    }),
+]
+
+sanity_pip_check = False
```
Comment (boegel) [inline easybuild/easyconfigs/s/Sphinx/Sphinx-1.8.3-foss-2019a-Python-3.7.2.eb:72]: @zarybnicky The error message in the tests is a bit confusing, but this should be `sanity_pip_check = True`.

We want to run `pip check` in the sanity check step to ensure that all required dependencies for Sphinx are installed.

File: easybuild/easyconfigs/s/Spyder/Spyder-4.1.5-foss-2019a-Python-3.7.2.eb [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    }),
+    ('yapf', '0.30.0', {
+        'checksums': ['3000abee4c28daebad55da6c85f3cd07b8062ce48e2e9943c8da1b9667d48427'],
+    }),
+    ('zipp', '3.1.0', {
+        'checksums': ['c599e4d75c98f6798c509911d08a22e6c021d074469042177c8c86fb92eefd96'],
+    }),
+]
+
+sanity_pip_check = False
```
Comment (boegel) [inline easybuild/easyconfigs/s/Spyder/Spyder-4.1.5-foss-2019a-Python-3.7.2.eb:278]: Same here, should be `sanity_pip_check = True`

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@zarybnicky I don't see a relation between Sphinx and Spyder, since Spyder doesn't depend on Sphinx (but installs Sphinx 3.2.1 as an extension to Spyder)? Maybe I'm overlooking something...

**Comment by zarybnicky** [comment]
@boegel As far as setup.py is concerned, sphinx is a dependency and part of the Help plugin: https://github.com/spyder-ide/spyder/blob/26227419e9c34a9a19a8316e0d4bd1ca55698013/setup.py#L246 But perhaps it should not be a standalone recipe but just a extension?

**Comment by zarybnicky** [comment]
Don't close - while everything works in the resulting Spyder build, pip check found a few inconsistencies. I guess I'm not a good dependency resolver if there are more than a hundred deps to juggle...

**Comment by boegel** [comment]
> @boegel As far as setup.py is concerned, sphinx is a dependency and part of the Help plugin: https://github.com/spyder-ide/spyder/blob/26227419e9c34a9a19a8316e0d4bd1ca55698013/setup.py#L246 But perhaps it should not be a standalone recipe but just a extension? I just updated the old Sphinx recipe to 2019a

If we have easyconfigs for `Sphinx` already, we indeed might as well update them and use them as a dependency for other stuff like `Spyder` (as opposed to installing `Sphinx` as an extension). But why the difference in Sphinx version, why not update the easyconfig to the latest `Sphinx` version?

**Comment by zarybnicky** [comment]
Will open a new one with just Spyder, don't want to confuse `eb --edit-pr` :man_shrugging:

**Pull request closed without merging**